### ISSUE TITLE
Media Capture and Playback Under Development? What does this mean for Core Audio?

### ISSUE THREAD

**Issue by DancinDirk** [Microsoft/WinObjC#47]
I was looking through the libraries and Core Audio is basically empty and the notes say Media Capture and Playback are still under development. Does this mean I'm going to have to wait for a bit before I begin work on making my App for Windows if CoreAudio is a major piece of it?

**Comment by s5msft** [comment]
@DancinDirk Maybe. If you want to directly reuse your iOS code for media capture/playback then you may have to hold off for a bit. Alternatively, if you're willing to #ifdef your way around a few things, then you can directly use relevant Windows APIs as appropriate.

The WOCCatalog sample shows you how to do some of this (admittedly not for CoreAudio though, just for video playback).

**Comment by DancinDirk** [comment]
Yeah, we spent a lot of time getting audio playback just right using core
audio so I'd like to reuse as much of that as possible, unless there's a
functionally equal windows version of coreAudio that could be easily
substituted in.

**Comment by colincornaby** [comment]
I'd love to see CoreAudio. In since CoreAudio is a C API, it would also be awesome to be able to share code with non Obj-C Windows apps. It would be nice to be able to share CoreAudio code without bringing in all the Obj-C support.

**Comment by hervevi** [comment]
Is there any simple way to play an mp3 file?

**Comment by s5msft** [comment]
@hervevi For simple media playback, you can use Xaml's MediaElement and embed it right in you app. 

See the following bit from one of the samples:
https://github.com/Microsoft/WinObjC/blob/master/samples/WOCCatalog/WOCCatalog/XamlViewController.m#L117

If you replace the given URL with something pointing to an mp3 file then you should get audio playback.

**Comment by hervevi** [comment]
@s5msft Thanks for this help, could you tell me how to read a mp3 embedded in the project? I tried to set the Uri to the name of my file, with or without the path, the program crashes (assertion failed). It does not find the file.

**Comment by hervevi** [comment]
I found a way to load the mp3 file that does not crash the program, but I can't hear anything :(
Here is my source code, called each time a mp3 should be played:

WXCMediaElement *mediaElement = [WXCMediaElement create];
mediaElement.autoPlay = YES;
mediaElement.source = [WFUri createUri: @"ms-appx:///mysound.mp3"];
[mediaElement play];

No crash, but nos sound too :(

**Comment by s5msft** [comment]
It's either that the Volume property isn't set/correct or that you need to add the MediaElement to the current View (WOCCatalog shows you how to do that). You should be able to set the width/height to 0 if you want no UI at all.

**Comment by hervevi** [comment]
Thanks a lot, it works now!
I now have to found a way to play a background music...
